lib/afc/streams.js

```diff
@@ -12,40 +14,33 @@ class AfcReadableFileStream extends stream.Readable {
     this._destroyed = false;
   }
 
-  async _read (size) {
-    try {
-      const data = await this._afcService.readFile(this._fileHandle, size);
-
-      if (this._destroyed) {
-        return;
-      }
-
-      if (_.isEmpty(data)) {
-        return this.push(null);
-      }
-
-      this.push(data);
-    } catch (e) {
-      if (this._autoDestroy) {
-        this.destroy(e);
-      } else {
-        this.emit('error', e);
-      }
-    }
+  _read (size) {
+    this._afcService.readFile(this._fileHandle, size)
+      .then((data) => {
+        if (!this._destroyed) {
+          this.push(_.isEmpty(data) ? null : data);
+        }
+      })
+      .catch((e) => {
+        if (this._autoDestroy) {
+          this.destroy(e);
+        } else {
+          this.emit('error', e);
+        }
+      });
   }
 
-  async _destroy (err, cb) {
+  _destroy (err, cb) {
     if (this._destroyed) {
       return;
     }
     this._destroyed = true;
     this.push(null);
-    try {
-      await this._afcService.closeFileHandle(this._fileHandle);
-    } catch (e) {
-      err = err || e;
-    }
-    cb(err);
+    this._afcService.closeFileHandle(this._fileHandle)
+      .catch((e) => {
+        err = err || e;
```

Comment:
I think that if both `err` and `e` exist here, `e` ought to be logged while `err` is thrown, so there is some visibility into the problem with closing the file handle.

Reply:
added logging